ADBDEV-4726-31 Check that lefttree is not null

dnskvlnk opened the pull request:
Check that lefttree is not null

ExecReScanNestLoop dereferences lefttree without checking that it's not null.
This patch adds Insist

RekGRpth commented on src/backend/executor/nodeNestloop.c:
```diff
@@ -585,6 +585,8 @@ ExecReScanNestLoop(NestLoopState *node)
 {
 	PlanState  *outerPlan = outerPlanState(node);
 
+	Assert(outerPlan);
```
May be add comment before, like: NestLoopState always has outerPlanState

dnskvlnk replied:
I think it's obvious that outerPlan shouldn't be null just because it's dereferenced here

RekGRpth replied:
Adding an assertion is not a solution to the problem. The static analyzer checks the release build.

dnskvlnk replied:
Then the only solution I see is replace assert with if (... == NULL) ereport. However, it doesn't seem like postgres-like solution, but it's the only way to stop the analyzer complining now and for all. If we try to add checks like this in arbitrary places pointed by the analyzer we will face the same problem in the future

RekGRpth replied:
Why report an error to the user if nothing can be done about it?

dnskvlnk replied:
You may use elog, it doesn't matter. My point is that the only way to stop the analyzer complining is to throw a runtime error at the very place a variable was derefernced rather then adding these check in the places pointer be the analyzer

RekGRpth commented:
Are there any foreign commits?

Commit:
Check that lefttree is not null

ExecReScanNestLoop dereferences lefttree without checking that it's not null.
This patch adds Insist
